examples/visionProcessing/ageDetection/ageDetection.cpp

```diff
@@ -0,0 +1,213 @@
+  std::vector<uint32_t> consumerPartitions;
+  for (auto i = consumerIndex; i <= highestPartitionNumber;
+       i += numberOfConsumers) {
+    consumerPartitions.push_back(i);
+  }
+  return consumerPartitions;
+}
+
+void run(const std::string &nodeName, const std::string &dagFileName,
+         uint32_t consumerIndex, uint32_t numberOfConsumers,
+         const std::string &protobufFile, const std::string &mlModel) {
+
+  AgeDetection ageDetection(protobufFile, mlModel);
+  ndn::Face face;
+
+  auto dagParser = iceflow::DAGParser::parseFromFile(dagFileName);
+  auto node = dagParser.findNodeByName(nodeName);
+  if (!node.downstream.empty()) {
+    std::string downstreamEdgeName;
+    for (const auto &downstream : node.downstream) {
+      if (downstream.id == "ad2agg") {
+        downstreamEdgeName = downstream.id;
+      }
+    }
+
+    if (downstreamEdgeName.empty()) {
+      std::cerr << "Error: Missing downstream target for age detection."
+                << std::endl;
+      return;
+    }
+
+    auto upstreamEdge = dagParser.findUpstreamEdges(node).at(0).second;
+    auto upstreamEdgeName = upstreamEdge.id;
+
+    auto consumerPartitions = createConsumerPartitions(
+        upstreamEdge.maxPartitions, consumerIndex, numberOfConsumers);
+
+    auto applicationConfiguration = node.applicationConfiguration;
+    auto saveThreshold =
+        applicationConfiguration.at("measurementsSaveThreshold")
+            .get<uint64_t>();
+
+    auto iceflow =
+        std::make_shared<iceflow::IceFlow>(dagParser, nodeName, face);
+
+    ::signal(SIGINT, signalCallbackHandler);
+    measurementHandler = new iceflow::Measurement(
+        nodeName, iceflow->getNodePrefix(), saveThreshold, "A");
+
+    auto prosumerCallback = [&iceflow, &ageDetection, &downstreamEdgeName](
+                                const std::vector<uint8_t> &encodedCroppedFace,
+                                iceflow::ProducerCallback producerCallback) {
+      auto pushDataCallback = [downstreamEdgeName, producerCallback](
+                                  std::vector<uint8_t> encodedAnalytics) {
+        producerCallback(downstreamEdgeName, encodedAnalytics);
+      };
+      ageDetection.ageDetection(encodedCroppedFace, pushDataCallback);
+    };
+
+    iceflow->registerProsumerCallback(upstreamEdgeName, prosumerCallback);
+    iceflow->repartitionConsumer(upstreamEdgeName, {0});
+    iceflow->run();
+  }
+}
+
+int main(int argc, const char *argv[]) {
+  if (argc != 6) {
+    std::cout << "usage: " << argv[0]
+              << " <application-dag-file><protobuf_binary> "
+                 "<ML-Model><instance-number> <number-of-instances>"
+              << std::endl;
+    return 1;
+  }
+
+  try {
+    std::string nodeName = "ageDetection";
+    std::string dagFileName = argv[1];
+    std::string protobufFile = argv[2];
+    std::string mlModel = argv[3];
+    int consumerIndex = std::stoi(argv[4]);
+    int numberOfConsumers = std::stoi(argv[5]);
```

Comment:
Since there is only one upstream partition, this could also be omitted here and the other places where this is being used.